docs/reference/userapi.rst

```diff
@@ -70,3 +75,129 @@ that best suits that file-format.
 .. autoclass:: imageio.core.format.Writer
     :inherited-members:
     :members: 
+
+.. _migration_from_v2:
+
+Migrating to ImageIO v3
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The primary novelty of the v3 API compared to the v2 API is that images are now
+treated as ndimages, a generalization of (flat) 2D images to N dimensions. This
+change allows us to design a much simpler API; however, comes with a few
+backwards incompatible changes.
+
+Avoiding API Migration for Legacy Scripts
+-----------------------------------------
+
+If you have an old script where you do not wish to migrate the API, you can
+explicitly import the old v2 API and avoid migration. Note that, while this will
+keep the old (v2) API, it will use on the new (v3) plugins and backends, which
+may contain other backwards incompatible changes. As such, you should prefer a
+full migration to v3, and should avoid using the v2 API in new code. To import
+the v2 API, depending on your script/code, use::
+
+    import imageio... as iio
+    import imageio... as imageio
+  
+
+Reading Images 
+--------------
+
+`iio.imread` can now return a ndimage instead of being limited to flat images.
+As such, `iio.volread` has merged into `iio.imread` and is now gone. Similarily,
+`iio.mimread` and `iio.mvolread` have merged into a new function called
+`iio.imiter`, which returns a generator that yields ndimages from a file in the
+order they appear. Further, the default behavior of `iio.imread` has changed,
+and it now returns the first ndimage (instead of the first flat image) if the
+image contains multiple images.
+
+To reproduce similar behavior to the v2 API behavior use one of the following
+snippets::
+
+    # img = iio.v2.imread(image_resource)
+    try:
+        img = iio.v3.imread(image_resource, index=None)[0]
+    except ValueError:
+        img = iio.v3.imread(image_resource, index=0)
+
+    # img = iio.v2.volread(image_resource)
+    img = iio.v3.imread(image_resource)
+    
+    # img = iio.v2.mimread(image_resource)
+    # img = iio.v2.mvolread(image_resource)
+    img = [im for im in iio.v3.imiter(image_resource)]
+
+Writing Images
+--------------
+
+Similar to reading images, the new `iio.imwrite` can handle ndimages.
+``iio.mimwrite``, ``iio.volwrite``, and ``iio.mvolwrite`` have all dissapeared.
+The same goes for their aliases ``iio.mimsave``, ``iio.volsave``,
+``iio.mvolsave``, and ``iio.imsave``. They are now all covered by
+``iio.imwrite``.
+
+To reproduce similar behavior to the v2 API behavior use one of the following
+snippets::
+
+    # img = iio.v2.imwrite(image_resource, ndimage)
+    # img = iio.v2.imsave(image_resource, ndimage)
```

Comment:
Maybe don't mention the aliases here to avoid verbosity.

Reply:
Will it be confusing for people that use the alias to not find it in the list? Or will us mentioning the aliases in the text above suffice?

Reply:
I think they'll get the point because the pattern is quite clear, and we mention the aliases above.